tractor/_runtime.py

```diff
@@ -702,6 +712,34 @@ async def _stream_handler(
                 #     if chan.uid == uid:
                 #         self._contexts.pop((uid, cid))
 
+                # NOTE: block this actor from acquiring the
+                # debugger-TTY-lock since we have no way to know if we
+                # cancelled it and further there is no way to ensure the
+                # lock will be released if acquired due to having no
+                # more active IPC channels.
+                if (
+                    _state.is_root_process()
+                ):
+                    pdb_lock = _debug.Lock
+                    log.cancel(
+                        f'{uid} is now blocked from debugger lock'
+                    )
+                    log.runtime(f"{uid} blocked from pdb locking")
+                    pdb_lock._blocked.add(uid)
+
+                    # if a now stale local task has the TTY lock still
+                    # we cancel it to allow servicing other requests for
+                    # the lock.
+                    if (
+                        pdb_lock._root_local_task_cs_in_debug
+                        and not pdb_lock._root_local_task_cs_in_debug.cancel_called
+                    ):
+                        log.warning(
+                            f'STALE DEBUG LOCK DETECTED FOR {uid}'
+                        )
+                        # TODO: figure out why this breaks tests..
```

Comment:
As mentioned above, so far it seems just putting the child on the block list is enough and avoids cancelling some actor that for wtv reason became IPC disconnected but still is using the debugger. My guess is there's a race between the scheduling of this cancel and a new request coming in?

Needs more tinkering/testing.